cluster.tf

```diff
@@ -0,0 +1,42 @@
+# GKE cluster
+resource "google_container_cluster" "primary" {
+  name     = local.resource_name
+  location = data.google_compute_zones.available.region
+
+  remove_default_node_pool = true
+  initial_node_count       = 1
+  network                  = data.ns_connection.network.outputs.vpc_name
+  subnetwork               = data.ns_connection.network.outputs.public_subnet_names[0]
+}
+
+# Service account to manage permission via IAM
+resource "google_service_account" "default" {
+  account_id   = local.resource_name
+  display_name = "${local.resource_name} Service Account"
+}
+
+# Managed Node Pool
+resource "google_container_node_pool" "primary_nodes" {
```

Comment:
We should configure the container node pool with an autoscaler with a variable for max nodes defaulting to 5. It would also be good to change the node count to minimum node count with a default of 1.

Reply:
Enabled auto scaler with min 1 node and max 5 nodes.

Reply:
Fantastic, can you update `README.md` with these variables.

Reply:
`README` file updated with input/output variables